### ISSUE TITLE
Unsuitable value for left operand

### ISSUE THREAD

**Issue by DevopsDood** [terraform-aws-modules/terraform-aws-vpc#278]
Not sure why I'm getting this error. I definitely have two public subnets defined. This is a copy paste of the example basically. I've played with some different options to no avail. 

Terraform 0.12.1

main.tf
```
variable "region" {}
variable "image_id" {}
variable "availability_zone_names" {}
variable "environment" {}
variable "profile" {
  default = "default"
}

provider "aws" {
  region  = var.region
  profile = var.profile
}

data "aws_security_group" "test" {
  name   = "test"
  vpc_id = module.vpc.vpc_id
}

resource "aws_instance" "web" {
  ami           = var.image_id
  instance_type = "t3.micro"
  count         = 2
  depends_on    = ["aws_internet_gateway.gw"]

  tags = {
    Name = "test"
  }
}

module "vpc" {
  source = "terraform-aws-modules/vpc/aws"

  name = "test-vpc"
  cidr = "10.0.0.0/16"

  azs            = var.availability_zone_names
  public_subnets = ["10.0.1.0/24", "10.0.0.0/24"]

  enable_nat_gateway   = true
  enable_dns_hostnames = true
  enable_dns_support   = true

  tags = {
    Terraform   = "true"
    Environment = var.environment
  }
}

resource "aws_internet_gateway" "gw" {
  vpc_id = module.vpc.vpc_id

  tags = {
    Name = "main"
  }
}

resource "aws_route_table" "r" {
  vpc_id = module.vpc.vpc_id
  count  = 2

  route {
    cidr_block = "0.0.0.0/24"
    gateway_id = aws_internet_gateway.gw.id
  }

  tags = {
    Name = "asfd"
  }
}
```

us-east-2.tfvars

```
region   = "us-east-2"
image_id = "ami-0c55b159cbfafe1f0"
availability_zone_names = [
  "us-east-2a",
  "us-east-2b",
]
environment = "test"
```

Error output:
```
Error: Invalid operand

  on .terraform/modules/vpc/terraform-aws-modules-terraform-aws-vpc-ed341bf/main.tf line 251, in resource "aws_subnet" "public":
 251:   count = var.create_vpc && length(var.public_subnets) > 0 && (false == var.one_nat_gateway_per_az || length(var.public_subnets)) >= length(var.azs) ? length(var.public_subnets) : 0

Unsuitable value for left operand: number required.


Error: Invalid operand

  on .terraform/modules/vpc/terraform-aws-modules-terraform-aws-vpc-ed341bf/main.tf line 251, in resource "aws_subnet" "public":
 251:   count = var.create_vpc && length(var.public_subnets) > 0 && (false == var.one_nat_gateway_per_az || length(var.public_subnets)) >= length(var.azs) ? length(var.public_subnets) : 0

Unsuitable value for right operand: bool required.
```

**Comment by bohdanyurov-gl** [comment]
@DevopsDood I've pushed a fix, please try my branch

**Comment by AshesOfOwls** [comment]
Ditto on this, I will test your branch @bohdanyurov-gl

**Comment by antonbabenko** [comment]
Thanks for opening this issue!

Already fixed in [v2.4.0](https://github.com/terraform-aws-modules/terraform-aws-vpc/releases/v2.4.0).

**Comment by RajeshGoud9** [comment]
While init am getting below error can u pls help
Error: Invalid operand

  on .terraform.lock.hcl line 53:
  (source code not available)

Unsuitable value for left operand: number required
